nixos/modules/hardware/printers.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, services, ... }:
+with lib;
+let
+  cfg = config.hardware.printers;
+  ensurePrinter = p: ''
+    ${pkgs.cups}/bin/lpadmin -p '${p.name}' -E ${optionalString (p.location != null) "-L '${p.location}'"} ${optionalString (p.description != null) "-D '${p.description}'"} -v '${p.deviceUri}' -m '${p.model}'
+  '';
+  ensureDefaultPrinter = name: ''
+    ${pkgs.cups}/bin/lpoptions -d '${name}'
+  '';
+
+  forbiddenChars  = str: filter (c: c == "#" || c == "/") (stringToCharacters str) == [];
+  # TODO: is it possible to combine the forbiddenChars with that regex?
```

Comment:
I'm 99% sure it's not possible because `builtins.match` is only POSIX regex, which doesn't support lookahead which I think would be needed for that.